pkg/server/stream/v3/stream.go

```diff
@@ -24,22 +24,43 @@ type DeltaStream interface {
 // StreamState will keep track of resource state per type on a stream.
 type StreamState struct {
 	// Indicates whether the original DeltaRequest was a wildcard LDS/RDS request.
-	Wildcard bool
+	wildcard bool
 
 	// ResourceVersions contains a hash of the resource as the value and the resource name as the key.
 	// This field stores the last state sent to the client.
-	ResourceVersions map[string]string
+	resourceVersions map[string]string
+
+	// indicates whether the object has beed modified since its creation
+	fresh bool
+}
+
+func (s *StreamState) GetResourceVersions() map[string]string {
```

Comment:
Do we need these wrappers? I'd be all for it if we had a lock we needed to hide, but I think we don't since resourceVersions will only be modified by one thread at a time so it just seems like extra boilerplate.

Reply:
I would also like to avoid having the wrappers. I added the wrapper for the  `SetResourceVersions ` method because I also set isFresh flag to false inside this method. The assumption is that we change the initial ResourceVersions only after we got the second request. Alternatively, we could manually set isFresh flag from the server code and keep accessing StreamState fields directly - this should make this PR much smaller, but I am not sure this is a better design. Let me know what you think. 

One more option is to keep `SetResourceVersions` wrapper but delete all other wrappers, but I think we should be consistent in how we access struct fields.

Reply:
Well from a straight performance perspective I think it would be faster to just access the fields directly. An exception applies to maps since those are always pointers but then we won't copy around bools and such. This is obviously minor so it might be worth trading off for readability. I'm good with leaving the wrappers for now